app/helpers/jobseekers/job_applications_helper.rb

```diff
@@ -0,0 +1,33 @@
+module Jobseekers::JobApplicationsHelper
+  def application_link_path
+    if jobseeker_application_status.nil?
+      new_jobseekers_job_job_application_path(@vacancy.id)
```

Comment:
Correct me if I'm very much mistaken, but I think Rails is clever enough that you can pass in the vacancy object itself, and it will work out the `id`.

Reply:
Same for the other paths in this method.

Reply:
I was wrong in this case - it tries to use the vacancy slug rather than the id.